src/hotspot/share/opto/inlinetypenode.cpp

```diff
@@ -445,7 +500,32 @@ void InlineTypeNode::store(GraphKit* kit, Node* base, Node* ptr, ciInstanceKlass
       if (ary_type != NULL) {
         decorators |= IS_ARRAY;
       }
-      kit->access_store_at(base, adr, adr_type, value, val_type, bt, decorators);
+      if (ft->bundle_size() > 1) {
+        int vec_len = ft->bundle_size();
+        BasicType elem_bt = ft->basic_type();
+
+        bool bundle_size_supported =
+          Matcher::match_rule_supported_vector(Op_StoreVector, vec_len, elem_bt) &&
+          Matcher::match_rule_supported_vector(VectorNode::replicate_opcode(elem_bt), vec_len, elem_bt);
+
+        // Set the vector length to  maximal supported vector length
+        // to allow graceful compilation exit at a later stage.
+        vec_len = bundle_size_supported ? vec_len : Matcher::max_vector_size(elem_bt);
+
+        // Handling for non-flattened case, with default InlineFieldMaxFlatSize of 128
+        // all the concrete vectors should be fully flattened.
+        value = value->bottom_type()->isa_vect() ? value : kit->gvn().transform(VectorNode::scalar2vector(value, vec_len, val_type, false));
+        assert(value->bottom_type()->isa_vect() && value->bottom_type()->is_vect()->length() == (uint)ft->bundle_size(), "");
```

Comment:
Is it possible that not all the multi-field values are the `value` here? For example, if the multi-fields in the InlineTypeNode have been changed before.

Reply:
I do follow your comment, multifield is not a value type to begin with, it the container VectorPayload* which is primitive class, for value types any change of field will result into creation of new InlineType, since values are immutable.